### PR TITLE
Rating Connection

### PR DESCRIPTION
**Pull request by davwys**
Add ratings, price estimates & break-evens to PropertyCard

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -23,4 +23,4 @@ pnpm-debug.log*
 *.sw?
 /.env.local
 /reports/coverage/
-/env.local
+/.env.local
```
Comment (thommann) [inline frontend/.gitignore]: already declared in line 24. redo, this is to make sure it isnt leaked if it isnt named properly

File: frontend/src/components/PropertyCard.vue [hunk 2]
```diff
@@ -112,18 +113,15 @@
...
 
       <q-item-label
         caption
         style="padding: 10px"
       >
-        Ratings are generated from aggregate data blabla.
-        <br>
-        <a href="https://www.google.ch">Learn how we determine ratings</a>
-        <!-- TODO -->
+        Ratings are generated from aggregate data of past sales of comparable properties, as well as prices and occupancy rates of nearby Airbnbs.
```
Comment (thommann) [inline frontend/src/components/PropertyCard.vue]: occupancy rates are not involved

File: frontend/src/components/PropertyCard.vue [hunk 3]
```diff
@@ -287,22 +294,12 @@
       >
         The break-even rating considers the following:
         <br>
-        - Est. maintenance cost of 10%
+        - Est. maintenance cost of 2%
```
Comment (thommann) [inline frontend/src/components/PropertyCard.vue]: per square feet per month

File: frontend/src/components/PropertyCard.vue [hunk 4]
```diff
@@ -287,22 +294,12 @@
...
         <br>
         - The stated capacity at the estimated price per night
         <br>
-        - Mortgage costs of 1.23% ($5678 p.a.)
-        <!-- TODO -->
+        - Mortgage costs of 2.5% at 80% mortgage ratio
```
Comment (thommann) [inline frontend/src/components/PropertyCard.vue]: interest rate of 3%, 75% of sale price as mortgage (ratio)

File: frontend/src/components/PropertyCard.vue [hunk 5]
```diff
@@ -431,12 +439,117 @@ export default {
...
+        console.log("Rating for", id);
+        // Get rating and apply to 1-10 range
+        if (result.data.data.saleById.score > 0) {
+          this.rating = Math.max(1, Math.round(result.data.data.saleById.score));
+        } else {
+          // Invalid rating
+          this.rating = 11;
+        }
+      });
+    },
```
Comment (thommann) [inline frontend/src/components/PropertyCard.vue:556]: fetch with initial property fetching. Just add score to the list of fields

File: frontend/src/components/menu/MenuDrawer.vue [hunk 6]
```diff
@@ -154,31 +152,31 @@
...
 
 <script>
-import AirbnbFilters from "./AirbnbFilters.vue";
+// import AirbnbFilters from "./AirbnbFilters.vue";
 
 export default {
   name: "MenuDrawer",
   components: {
-    AirbnbFilters,
+    // AirbnbFilters,
```
Comment (thommann) [inline frontend/src/components/menu/MenuDrawer.vue:179]: what are these filters?

File: frontend/src/pages/MainPage.vue [hunk 7]
```diff
@@ -109,7 +109,11 @@ export default {
      * Upon clicking a marker, mark it as selected
      * @param {Object} marker - the marker that was clicked, having id, lat and lng
      */
-    onMarkerClick(marker) {
+    async onMarkerClick(marker) {
+      // Briefly switch between selected properties
+      this.$store.commit("setSelectedCoordinates", { selected_coordinates: null });
+      await new Promise((resolve) => setTimeout(resolve, 10));
```
Comment (thommann) [inline frontend/src/pages/MainPage.vue]: Why is this here?

**Review by thommann — CHANGES_REQUESTED** [review]
remove .DS_Store

File: frontend/src/components/PropertyCard.vue (hunk 5 above)
Comment (davwys) [reply frontend/src/components/PropertyCard.vue:556]: The idea was to fetch this afterwards, since score calculation is kind of slow - or does it take the same time regardless of whether score is fetched?

File: frontend/src/components/menu/MenuDrawer.vue (hunk 6 above)
Comment (davwys) [reply frontend/src/components/menu/MenuDrawer.vue:179]: Various filters for which Airbnbs to show (e.g. by price, etc.)
I think we can leave them out and add them when we have time to spare, since I don't think this filtering is implemented on the backend side.

File: frontend/src/pages/MainPage.vue (hunk 7 above)
Comment (davwys) [reply frontend/src/pages/MainPage.vue]: This was a not so ideal workaround for another issue, fixed properly now
